src/CodeSigning/CodeSigning/Commands/GetAzureCodeSigningRootCert.cs

```diff
@@ -0,0 +1,127 @@
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.CodeSigning.Helpers;
+using Microsoft.Azure.Commands.CodeSigning.Models;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Azure.Commands.CodeSigning
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzurePrefix + "CodeSigningRootCert", DefaultParameterSetName = ByAccountProfileNameParameterSet)]
+    [OutputType(typeof(string))]
+    public class GetAzureCodeSigningRootCert : CodeSigningCmdletBase
+    {
+        #region Parameter Set Names
+
+        private const string ByAccountProfileNameParameterSet = "ByAccountProfileNameParameterSet";
+        private const string ByMetadataFileParameterSet = "ByMetadataFileParameterSet";
+
+        #endregion
+
+        #region Input Parameter Definitions
+
+        /// <summary>
+        /// Account Profile name
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The account name of Azure CodeSigning.")]
+        [ValidateNotNullOrEmpty]
+        public string AccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 1,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The certificate profile name of Azure CodeSigning account.")]
+        [ValidateNotNullOrEmpty()]
+        public string ProfileName { get; set; }
+        [Parameter(Mandatory = true,
+            Position = 2,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The endpoint url used to submit request to Azure CodeSigning.")]
+        public string EndpointUrl { get; set; }
+
+
+        /// <summary>
+        /// Metadata File Path
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ByMetadataFileParameterSet,
+             ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Metadata File path. Cmdlet constructs the FQDN of an account profile based on the Metadata File and currently selected environment.")]
+        [ValidateNotNullOrEmpty]
+        public string MetadataFilePath { get; set; }
+
+        [Parameter(Mandatory = true,
+          Position = 3,
+          ParameterSetName = ByAccountProfileNameParameterSet,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "Downloaded Root Cert file full path, including file name")]
+        [Parameter(Mandatory = true,
+          Position = 3,
```

Comment:
There is no position 1 and position 2 in `ByMetadataFileParameterSet`. Also, we don't suggest to use position more than 3.

Reply:
> I fixed the order for the set highlighted, good catch. Concerning the suggestion, do need that bunch of parameters for invoking, any suggestion or guidance on viable alternatives?